app/src/main/java/com/termux/app/TermuxService.java

```diff
@@ -618,20 +658,23 @@ public synchronized int removeTermuxSession(TerminalSession sessionToRemove) {
     /** Callback received when a {@link TermuxSession} finishes. */
     @Override
     public void onTermuxSessionExited(final TermuxSession termuxSession) {
-        if (termuxSession != null) {
+        boolean isTermuxSession = termuxSession != null;
```

Comment:
These and all the other if conditions extracted to boolean are useless and server no purpose, using an if condition already implies that one or more conditions will be supplied in it. Do not do this. If there are multiple conditions that are too complex, they can be split on multiple lines, in some case extraction to boolean is indeed fine.